src/tokens/ERC1155.sol

```diff
@@ -59,20 +69,25 @@ abstract contract ERC1155 {
         uint256 amount,
         bytes calldata data
     ) public virtual {
-        require(msg.sender == from || isApprovedForAll[from][msg.sender], "NOT_AUTHORIZED");
+        if (msg.sender != from && !isApprovedForAll[from][msg.sender]) {
+            revert NOT_AUTHORIZED();
+        }
 
         balanceOf[from][id] -= amount;
         balanceOf[to][id] += amount;
 
         emit TransferSingle(msg.sender, from, to, id, amount);
 
-        require(
-            to.code.length == 0
-                ? to != address(0)
-                : ERC1155TokenReceiver(to).onERC1155Received(msg.sender, from, id, amount, data) ==
-                    ERC1155TokenReceiver.onERC1155Received.selector,
-            "UNSAFE_RECIPIENT"
-        );
+        if (
+            (
+                to.code.length == 0
+                    ? to != address(0)
+                    : ERC1155TokenReceiver(to).onERC1155Received(msg.sender, from, id, amount, data) ==
+                        ERC1155TokenReceiver.onERC1155Received.selector
+            ) == false
```

Comment:
== false -> !

Reply:
The syntax in v7 changes anywho so marking resolved